app/services/scheduler_service.rb

```diff
@@ -0,0 +1,154 @@
+class SchedulerService
+  require "google/apis/calendar_v3"
+  require "googleauth"
+  require "googleauth/stores/file_token_store"
+  require "date"
+  require "fileutils"
+
+  attr_reader :current_user, :weekdays, :hour_scheduled_times
+
+  def initialize(current_user)
+    @current_user = current_user
+    @weekdays = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]
+    @hour_scheduled_times = [("08:01".."08:59"), ("09:01".."09:59"),("10:01".."10:59"),("11:01".."11:59"),("12:01".."12:59"),( "13:01".."13:59"),( "14:01".."14:59"), ("15:01".."15:59"), ("16:01".."16:59"), ("17:01".."17.59"), ("18:01".."18:59"),( "19:01".."19:59"), ("20:01".."20:59")]
+  end
+
+  # sets up initial Google Calendar Service with current user's google token
+  def get_calendar_service
+    service ||= Google::Apis::CalendarV3::CalendarService.new
+    service.client_options.application_name = "Treat Yo Self"
+    service.authorization = @current_user.google_token
+    service
+  end
+
+  # Uses Google Service to find a list of user events within current week time period
+  def find_user_events
+    calendar_id = "primary"
+    response = get_calendar_service.list_events( calendar_id,
+                                   time_min: DateTime.now.rfc3339,
+                                   time_max: (DateTime.now + 1.week).rfc3339,
+                                   single_events: true,
+                                   order_by: "startTime" )
+  end
+
+  # Sets available times from 8 a.m. - 8 p.m. as a string and puts that
```

Comment:
Thank you for adding in the comments in this file. Will be worth having everyone read through it before Wednesday and then possibly removing it?

Reply:
This is a great solution. Thank you for the comments to help with understanding the code.